pkg/auth/providers/github/github_provider_test.go

```diff
@@ -0,0 +1,230 @@
+package github
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/rancher/norman/types"
+	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type fakeTokensManager struct {
+	getSecretFunc               func(userID string, provider string, fallbackTokens []*v3.Token) (string, error)
+	isMemberOfFunc              func(token v3.Token, group v3.Principal) bool
+	createTokenAndSetCookieFunc func(userID string, userPrincipal v3.Principal, groupPrincipals []v3.Principal, providerToken string, ttl int, description string, request *types.APIContext, userExtraInfo map[string][]string) error
+}
+
+func (m *fakeTokensManager) GetSecret(userID string, provider string, fallbackTokens []*v3.Token) (string, error) {
+	if m.getSecretFunc != nil {
+		return m.getSecretFunc(userID, provider, fallbackTokens)
+	}
+	return "", nil
+}
+
+func (m *fakeTokensManager) IsMemberOf(token v3.Token, group v3.Principal) bool {
+	if m.isMemberOfFunc != nil {
+		return m.isMemberOfFunc(token, group)
+	}
+	return false
+}
+
+func (m *fakeTokensManager) CreateTokenAndSetCookie(userID string, userPrincipal v3.Principal, groupPrincipals []v3.Principal, providerToken string, ttl int, description string, request *types.APIContext, userExtraInfo map[string][]string) error {
+	if m.createTokenAndSetCookieFunc != nil {
+		return m.createTokenAndSetCookieFunc(userID, userPrincipal, groupPrincipals, providerToken, ttl, description, request, userExtraInfo)
+	}
+	return nil
+}
+
+func TestSearchPrincipals(t *testing.T) {
+	var userOrgs, orgTeams, searchUsersAll, searchUsersGroup, searchUsersUser []byte
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch path := r.URL.Path; path {
+		case "/api/v3/user/orgs":
+			w.Write(userOrgs)
+		case "/api/v3/orgs/devorg/teams":
+			w.Write(orgTeams)
+		case "/api/v3/search/users":
+			q := r.URL.Query().Get("q")
+			if strings.Contains(q, " type:org") {
+				w.Write(searchUsersGroup)
+			} else if strings.Contains(q, " type:user") {
+				w.Write(searchUsersUser)
+			} else {
+				w.Write(searchUsersAll)
+			}
+		default:
+			t.Errorf("Unexpected client call %s", path)
+		}
+	}))
+	defer srv.Close()
+
+	srvURL, err := url.Parse(srv.URL)
+	if err != nil {
+		t.Fatal(err)
```

Comment:
Is the use of Fatal instead of Error here intentional?

Reply:
answering myself....seems reasonable and deservedly intentional here.

Reply:
Yes, if it failed for whatever reason (most unlikely though) there is no reason to continue.